library/core/src/mem/mod.rs

```diff
@@ -1011,16 +1011,48 @@ pub fn copy<T: Copy>(x: &T) -> T {
 /// Interprets `src` as having type `&Dst`, and then reads `src` without moving
 /// the contained value.
 ///
-/// This function will unsafely assume the pointer `src` is valid for [`size_of::<Dst>`][size_of]
-/// bytes by transmuting `&Src` to `&Dst` and then reading the `&Dst` (except that this is done
-/// in a way that is correct even when `&Dst` has stricter alignment requirements than `&Src`).
-/// It will also unsafely create a copy of the contained value instead of moving out of `src`.
+/// `transmute_copy` is semantically equivalent to a bitwise copy of one type
+/// into another. It copies the bits from the source value into the destination
+/// value. Unlike [`transmute`], it preserves the original. Note that
+/// destination is passed by-value, which means if `Dst` contains padding,
+/// that padding is *not* guaranteed to be preserved by `transmute`.
 ///
 /// It is not a compile-time error if `Src` and `Dst` have different sizes, but it
 /// is highly encouraged to only invoke this function where `Src` and `Dst` have the
 /// same size. This function triggers [undefined behavior][ub] if `Dst` is larger than
 /// `Src`.
 ///
+/// # Safety
+///
+/// Behavior is [undefined][ub] if any of the following conditions are violated:
+///
+/// * The reference `src` must be valid for [`size_of::<Dst>`][size_of] bytes. In other words,
+///   [`size_of::<Src>`][size_of] must be greater than or equal to [`size_of::<Dst>`][size_of].
+///
+/// * All the safety conditions of [`transmute`] must hold:
+///
+///     * Both the argument and the result must be [valid](../../nomicon/what-unsafe-does.html)
```

Comment:
I think this sub-list doesn't make sense because it's not exhaustive. The link to `transmute` itself is already sufficient.
Or maybe the sentences below could be rewritten as additional bullet points.

Having a mix of paragraphs and bullet points is dangerous as a reader might think one is the summary of the other.